### ISSUE TITLE
Train loss is nan but valid loss is non-nan

### ISSUE THREAD

**Issue by zafarali** [Lasagne/Lasagne#529]
Hi everyone!

I'm trying to train a convolution neural network, my output during the training phase looks like this:

```
data shapes: (1419620, 32)
event shapes (1419620, 1)

/Users/zaf/anaconda/lib/python2.7/site-packages/sklearn/cross_validation.py:417: Warning: The least populated class in y has only 1 members, which is too few. The minimum number of labels for any class cannot be less than n_folds=5.
  % (min_labels, self.n_folds)), Warning)
# Neural Network with 18417 learnable parameters

## Layer information

  #  name      size
---  --------  ------
  0  layer1    32x64
  1  conv1     6x64
  2  conv2     3x63
  3  maxpool1  3x15
  4  dense1    300
  5  drop2     300
  6  output    12

/Users/zaf/anaconda/lib/python2.7/site-packages/numpy/core/_methods.py:59: RuntimeWarning: Mean of empty slice.
  warnings.warn("Mean of empty slice.", RuntimeWarning)
  epoch    train loss    valid loss    train/val    valid acc  dur
-------  ------------  ------------  -----------  -----------  -----
      1           nan       4.51024          nan      0.00000  0.05s
      2           nan       4.51024          nan      0.00000  0.04s
      3           nan       4.51024          nan      0.00000  0.04s
      4           nan       4.51024          nan      0.00000  0.04s
      5           nan       4.51024          nan      0.00000  0.03s
      6           nan       4.51024          nan      0.00000  0.05s
      7           nan       4.51024          nan      0.00000  0.04s
      8           nan       4.51024          nan      0.00000  0.03s
      9           nan       4.51024          nan      0.00000  0.05s
     10           nan       4.51024          nan      0.00000  0.04s
     11           nan       4.51024          nan      0.00000  0.04s
     12           nan       4.51024          nan      0.00000  0.04s
```

This is my network:
(Note: **Layer factory just ensures that I initialize all the layers in the right way
```python
    layer_list = [
        LF(layers.InputLayer, shape=(None, N_ELECTRODES, TIME_POINTS)), 
        LF(layers.Conv1DLayer, 'conv1', num_filters=6, filter_size=5, nonlinearity=None, pad='same'),
        LF(layers.Conv1DLayer, 'conv2', num_filters=3, filter_size=2),
        LF(layers.MaxPool1DLayer, 'maxpool1', pool_size=4),
        LF(layers.DenseLayer, 'dense1', num_units=300),
        LF(layers.DropoutLayer, 'drop2', p=0.5),

        LF(layers.DenseLayer, layer_name="output", num_units=N_EVENTS, nonlinearity=lasagne.nonlinearities.softmax)
    ]
    
    nnet =  NeuralNet(
        layers = layer_list,
        batch_iterator_train = batch_iter_train,
        batch_iterator_test = batch_iter_test,
        max_epochs=max_epochs,
        verbose=1000000,
        update = adagrad, 
        update_learning_rate = 0.01,
        **LF.kwargs
        )
```

Any idea why this would happen and where I should look  to go bug hunting?

Zaf

**Comment by f0k** [comment]
> Any idea why this would happen and where I should look to go bug hunting?

Maybe it's got something to do with your "RuntimeWarning: Mean of empty slice."? Is there some empty training batch somewhere and you're dividing by its length?

Anyway, let's move the discussion to our mailing list: https://groups.google.com/forum/#!forum/lasagne-users (i.e., please repost the question there if my hint doesn't solve it already.)
The github issue tracker is meant for bug reports and feature discussions. Thank you!

**Comment by zafarali** [comment]
Thanks! This was indeed the problem, I have a massive class imbalance and
it turns out that I was always supplying the same y values.
